### PR TITLE
tests: update the tests to start using tests.systemd tool instead of systemd.sh

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This is the second part of the migration from systemd.sh helper to tests.systemd tool

This change removes the systemd.sh helper

### CODE REVIEW

**Review by Meulengracht — APPROVED** [review]

**Comment by ernestl** [comment]
File `tests/main/deb-restart-behavior/task.yaml` still reference `. "$TESTSLIB"/systemd.sh`

File: tests/main/snap-confine-from-core/task.yaml [hunk 1]
```diff
@@ -14,9 +14,6 @@ restore: |
     chmod 4755 /usr/lib/snapd/snap-confine
 
 execute: |
-    # shellcheck source=tests/lib/systemd.sh
-    . "$TESTSLIB/systemd.sh"
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/main/snap-confine-from-core/task.yaml:18]: nice catch, thanks for the review

**Review by ernestl — APPROVED** [review]
Looks good

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**